media/rtm_change_upload_directory.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * This snippet allows you to change upload directory to /wp-content/uploads/rtMedia/users/userid/postid/filename.
+ * The media must be uploaded through rtMedia uploader (calling `rtmedia_filter_upload_dir` filter).
+ * This code will only work if the media is being uploaded from the post, i.e. if `get_the_ID()` returns the valid post ID, it won't work if you upload from the profile.
+ *
+ * All your previously uploaded media files won't be affected by this code, but new media files will be uploaded to specified directory.
+ */
+
+// Check whether function exists or not.
+if ( ! function_exists( 'rtmedia_set_upload_path_to_post_id' ) ) {
+
+	/**
+	 * This filter is called when a media is being uploaded through rtMedia upload mechanism.
+	 * This filter is used to change the upload directory.
+	 */
+	add_filter( 'rtmedia_filter_upload_dir', 'rtmedia_set_upload_path_to_post_id', 10, 1 );
+
+	/**
+	 * This function will alter the upload url, path and subdir parameters
+	 *
+	 * @param $param array : upload parameters, ( path, url, ... )
+	 *
+	 * @return array
+	 */
+	function rtmedia_set_upload_path_to_post_id( $param ) {
+
+		// Change parameters only if the URL is from rtMedia and `get_the_ID()` returns valid post ID.
+		if ( get_the_ID() && stripos( $param['url'], 'rtmedia' ) !== false ) {
```

Comment:
Don't call `get_the_ID()` multiple times. Instead take value in variable then use. Also use `! empty()` for post ID.